### PR TITLE
Metrics: Autocollected Metric Extraction now uses new aggregation APIs

### PR DESCRIPTION
**Pull request by macrogreg**
This is PR number 2 that addresses https://github.com/Microsoft/ApplicationInsights-dotnet/issues/806

One last PR is coming to remove the no longer used internal APIs.

### CODE REVIEW

File: src/Microsoft.ApplicationInsights/Extensibility/Implementation/Metrics/DependencyMetricsExtractor.cs [hunk 1]
```diff
@@ -88,19 +97,20 @@ public int MaxDependencyTypesToDiscover
...
         }
 
         /// <summary>
-        /// Initializes the internal metrics trackers based on settings.
+        /// Pre-initialize this extractor.
         /// </summary>
-        /// <param name="metricManager">The <c>MetricManager</c> to be used for creating and sending the metrics by this extractor.</param>
-        public void InitializeExtractor(MetricManagerV1 metricManager)
+        /// <param name="metricTelemetryClient">The <c>TelemetryClient</c> to be used for sending extracted metrics.</param>
+        public void InitializeExtractor(TelemetryClient metricTelemetryClient)
```
Comment (Dmitry-Matveev) [inline src/Microsoft.ApplicationInsights/Extensibility/Implementation/Metrics/DependencyMetricsExtractor.cs]: This looks like a public API contract change from `void init(MetricManager)` to `void init(TelemetryClient)` - is this a breaking change or this class was hidden in 2.6?

File: src/Microsoft.ApplicationInsights/Extensibility/Implementation/Metrics/RequestMetricsExtractor.cs [hunk 2]
```diff
@@ -1,46 +1,48 @@
...
         {
         }
 
         public string ExtractorName { get; } = "Requests";
 
-        public string ExtractorVersion { get; } = "1.0";
+        public string ExtractorVersion { get; } = "1.1";
 
-        public void InitializeExtractor(MetricManagerV1 metricManager)
+        public void InitializeExtractor(TelemetryClient metricTelemetryClient)
```
Comment (Dmitry-Matveev) [inline src/Microsoft.ApplicationInsights/Extensibility/Implementation/Metrics/RequestMetricsExtractor.cs]: Same question here.

File: src/Microsoft.ApplicationInsights/Extensibility/Implementation/Metrics/DependencyMetricsExtractor.cs (hunk 1 above)
Comment (macrogreg) [reply src/Microsoft.ApplicationInsights/Extensibility/Implementation/Metrics/DependencyMetricsExtractor.cs]: The class is internal, so no breaking change. Internal classes have public methods if the method represents the contract of the class, but the class is not intended for public consumption.

File: src/Microsoft.ApplicationInsights/Extensibility/Implementation/Metrics/RequestMetricsExtractor.cs (hunk 2 above)
Comment (macrogreg) [reply src/Microsoft.ApplicationInsights/Extensibility/Implementation/Metrics/RequestMetricsExtractor.cs]: No breaking change. Same answer. :)

**Pull request merged**